core-java-modules/core-java-concurrency-collections-2/src/main/java/com/baeldung/concurrent/lock/SingleLock.java

```diff
@@ -0,0 +1,44 @@
+package com.baeldung.concurrent.lock;
+
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantLock;
+
+import com.google.common.base.Supplier;
+
+public class SingleLock extends ConcurrentAccessExperiment {
+    ReentrantLock lock;
+
+    public SingleLock() {
+        lock = new ReentrantLock();
+    }
+
+    protected synchronized Supplier<?> putSupplier(Map<String,String> map, int key) {
+        return (()-> {
+            boolean done = false;
+            try {
+                while(!done) {
```

Comment:
Why not simply use `lock.lock()`? That seems to be the default suggestion from [the JavaDoc](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/ReentrantLock.html).

Reply:
I was trying to keep both *Lock classes the more alike. Also, without this, we should stop and resume tasks that couldn't acquire the lock: that would be out of scope for this tutorial.

Reply:
done, but I had to replicate the behavior in both *Lock classes